nautobot/utilities/testing/__init__.py

```diff
@@ -32,15 +42,73 @@
     "ModelTestCase",
     "ModelViewTestCase",
     "ViewTestCases",
+    "run_job_for_testing",
 )
 
 
+def run_job_for_testing(job, data=None, commit=True, username="test-user", request=None):
+    """Provide a common interface to run Nautobot jobs as part of unit tests.
+
+    Args:
+      job (Job): Job model instance (not Job class) to run
+      data (dict): Input data values for any Job variables.
+      commit (bool): Whether to commit changes to the database or rollback when done.
+      username (str): Username of existing or to-be-created User account to own the JobResult. Ignored if `request.user`
+        exists.
+      request (HttpRequest): Existing request (if any) to own the JobResult.
+
+    Returns:
+      JobResult: representing the executed job
+    """
+    if data is None:
+        data = {}
+
+    # If the request has a user, ignore the username argument and use that user.
+    if request and request.user:
+        user_instance = request.user
+    else:
+        user_model = get_user_model()
+        user_instance, _ = user_model.objects.get_or_create(username=username, is_superuser=True, password="password")
+    job_result = JobResult.objects.create(
+        name=job.class_path,
+        obj_type=ContentType.objects.get_for_model(Job),
+        user=user_instance,
+        job_model=job,
+        job_id=uuid.uuid4(),
+    )
+
+    @contextmanager
+    def _web_request_context(user):
+        if request:
+            yield request
+        else:
+            yield web_request_context(user=user)
+
+    with _web_request_context(user=user_instance) as request:
+        run_job(data=data, request=request, commit=commit, job_result_pk=job_result.pk)
+    return job_result
+
+
 @tag("unit")
 class TransactionTestCase(_TransactionTestCase):
     """
     Base test case class using the TransactionTestCase for unit testing
     """
 
+    # 'job_logs' is a proxy connection to the same (default) database that's used exclusively for Job logging
+    if "job_logs" in settings.DATABASES:
+        databases = ("default", "job_logs")
+
+    def setUp(self):
+        """Provide a clean, post-migration state before each test case.
+
+        django.test.TransactionTestCase truncates the database after each test runs. We need at least the default
+        statuses present in the database in order to run tests."""
+        super().setUp()
+
+        # Re-populate status choices after database truncation by TransactionTestCase
+        populate_status_choices(apps, None)
```

Comment:
Out of curiosity - would it also work, and perhaps even be more correct, to override `tearDown` and do this there instead?

Reply:
I just tested this and unfortunately it does not. The reason for this is that `TransactionTestCase` has a `_post_teardown` method that is called after `tearDown`:

```python
def _post_teardown(self):
    """
    Perform post-test things:
    * Flush the contents of the database to leave a clean slate. If the
      class has an 'available_apps' attribute, don't fire post_migrate.
    * Force-close the connection so the next test gets a clean cursor.
    """
```

As a consequence the newly re-created Statuses would always get immediately truncated.

Reply:
OK - thank you for trying it!